src/lib/Live/Eval.hs

```diff
@@ -56,77 +97,207 @@ data CacheState = CacheState
 emptyCache :: CacheState
 emptyCache = CacheState 0 1 mempty
 
-runDriver :: DriverCfg -> TopStateEx -> Actor SourceContents
+class (Monad m, MonadIO m) => Driver m where
+  askOptions :: m EvalConfig
+  askResultsOutput :: m (PChan RFragment)
+  askSelf :: m (PChan DriverEvent)
+  getTopState :: m (WithId TopStateEx)
+  putTopState :: WithId TopStateEx -> m ()
+  -- Resets the evaluation state to initial, from the given TopStateEx.
+  -- Returns the old top state and the old worker spec, for reuse
+  refresh :: TopStateEx -> m (WithId TopStateEx, WorkerSpec)
+  -- Get the work chunk we are waiting for, if any
+  getWorkingBlock :: m (Maybe (WithId SourceBlock))
+  -- Run the action if there is no worker
```

Comment:
And what if there is a worker?

Reply:
Also, how about renaming this to `whenAbsentWorker`? That's more consistent with `Control.Monad.when`

Reply:
Done.